src/features/events/store.ts

```diff
@@ -515,32 +515,50 @@ const eventsSlice = createSlice({
 
 function addEventToState(state: EventsStoreSlice, events: ZetkinEvent[]) {
   events.forEach((event) => {
-    const dateStr = event.start_time.slice(0, 10);
-
-    if (!state.eventsByDate[dateStr]) {
-      state.eventsByDate[dateStr] = remoteList();
+    const newDateStr = event.start_time.slice(0, 10);
+    if (!state.eventsByDate[newDateStr]) {
+      state.eventsByDate[newDateStr] = remoteList();
     }
 
     const eventListItem = state.eventList.items.find(
       (item) => item.id == event.id
     );
-    const eventByDateItem = state.eventsByDate[dateStr].items.find(
-      (item) => item.id == event.id
-    );
 
     if (eventListItem) {
       eventListItem.data = { ...eventListItem.data, ...event };
     } else {
       state.eventList.items.push(remoteItem(event.id, { data: event }));
     }
 
-    if (eventByDateItem) {
-      eventByDateItem.data = { ...eventByDateItem.data, ...event };
-      eventByDateItem.mutating = [];
-    } else {
-      state.eventsByDate[dateStr].items.push(
+    let targetedEventDateToUpdate: string | undefined = undefined;
+    for (const date in state.eventsByDate) {
+      const item = state.eventsByDate[date].items.find(
+        (item) => item.id == event.id
+      );
+      if (item) {
+        targetedEventDateToUpdate = item.data?.start_time.slice(0, 10);
+      }
+    }
+
+    if (targetedEventDateToUpdate) {
+      // When updating an event date, remove the old date's event from state.eventsByDate and push that event to the new date.
+      state.eventsByDate[targetedEventDateToUpdate].items = state.eventsByDate[
+        targetedEventDateToUpdate
+      ].items.filter((item) => item.data?.id !== event.id);
+      state.eventsByDate[newDateStr].items.push(
         remoteItem(event.id, { data: event })
       );
+    } else {
+      //This is to check if the event already exists. In some places, addEventToState might be called more than once.
+      if (
+        !state.eventsByDate[newDateStr].items.some(
+          (item) => item?.data?.id === event.id
```

Comment:
I think this is very hard to read when it is inside an `if`, it would make it easier for me to read if the return value was saved, and that was used inside the `if`

Reply:
Sometimes it is quite hard for me to think what is better in this kind of case. I've been also thinking about it, and thought maybe saving this in a variable would not necessary since it takes the memory and only used in one place. But for making code more readable, I think I should have assigned it to variable. Thank you Ziggi! Helped me a lot! 👍